### PR TITLE
Fixes #13294 - Add qoutes to search parameter in generate_links_for h…

### PR DESCRIPTION
**Pull request by shlomizadok**
…elper method

### CODE REVIEW

**Comment by dLobatog** [comment]
@shlomizadok Nice addition, thanks! Mind to add one or two tests for this method? This seems easy to break without anyone noticing.

**Comment by shlomizadok** [comment]
@dLobatog - where shall I add such tests? Integration / functional?

**Comment by dLobatog** [comment]
@shlomizadok Just a regular unit test for the helper should do, notice we have a few already in `test/unit/helpers`

File: test/unit/helpers/application_helper_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class ApplicationHelperTest < ActionView::TestCase
+  include ApplicationHelper
+
+  def test_generate_link_for
+    proxy = FactoryGirl.create(:smart_proxy)
+    subnet = FactoryGirl.create(:subnet, name: 'My subnet')
```
Comment (dLobatog) [inline test/unit/helpers/application_helper_test.rb:8]: This hash should be 1.9.3 style, but it's OK, I'll fix it on merge

**Comment by dLobatog** [comment]
Merged as 2aee8c9941d4507e2065ae57d6305b24c764cfab, thanks @shlomizadok !

**Pull request closed without merging**